### DISCUSSION TITLE
FormState isValid is false but errors object is empty

### DISCUSSION THREAD

**Discussion by mimameid** [react-hook-form/react-hook-form#6858] [Q&A]
Hello everyone,

I'm running into a weird issue, where I can't tell if the behavior is correct and depending on the answer, how to achieve what I want.

The validation is running `onBlur` as expect and determined in the`mode` field of `useForm`. I can't update the UI though, since the error is not reflected in the `formState.errors` field.

What am I doing wrong? 

**Update:**

After digging a bit deeper into what is going on, since I couldn't quite understand, what I was doing wrong. I figured, that validation is working differently for `fieldArrays` compared to field level input. In both validation is run, though for `fieldArrays` the errors aren't reflected in the formstate immediately, but only after submit. While this works for field level input. You can find it here:

https://codesandbox.io/s/sweet-rain-h38ys

vs

https://codesandbox.io/s/zen-haze-52ukk

**Comment by bluebill1049** [comment]
we don't return the entire form `errors` and focus on a user interactive field, to save computation and re-render. This RFC maybe the answer you after: https://github.com/react-hook-form/react-hook-form/discussions/6472

**Reply by mimameid** [reply]
I'm not quite sure if I understand. Apparently the validation is running `onBlur`. How is not reflecting the errors that the validation detected boosting the performance?

**Reply by JamesWong1999** [reply]
```tsx
mode: "onChange",
reValidateMode: "onBlur"
```

according to your codesandbox your validation is run `onChange` then switched to `onBlur` afater submitting.

`errors` are referred to the field level with which field you are interactive with.
`isValid` refers to the form level.

```
for (let weekday of Object.keys(weekdays)) {
  // eslint-disable-next-line react-hooks/rules-of-hooks
  fieldArrays[weekday] = useFieldArray({
    control,
    name: weekday
  });
}
```

I don't think this above is correct usage either.

**Reply by mimameid** [reply]
Oh, the `reValidateMode` was only there for testing. My bad.

I understand what you are saying regarding `errors` referring to field level and `isValid` referring to form level. What I don't understand is, how it is related to the question. As far as I understand, the validation for each `fieldArray` is executed `onBlur`. Could you elaborate?

Regarding the usage of `useFieldArray`:

Why is it not used correctly and more importantly, how would it be correct? 

Regards

**Comment by mimameid** [comment]
@bluebill1049  After looking into it, I noticed, that validation is working differently depending on the input being field level or not. If this is on purpose, would it make sense to add it to the documentation somewhere?
